make/validate.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+#
+# Helper script to validate terraform input (avoid disaster).
+BASE_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+export BASE_DIR
+
+# shellcheck source=/dev/null
+source "${BASE_DIR}/helpers.sh"
+# shellcheck source=/dev/null
+source "${BASE_DIR}/variables.sh"
+
+ACTION=${1}
+
+# Validate input values.
+SUPPORTED_ACTIONS=("validate_input")
+if ! [[ " ${SUPPORTED_ACTIONS[*]} " == *" ${ACTION} "* ]]; then
+  print_error "Invalid action. Must be one of '${SUPPORTED_ACTIONS[*]}'."
+  exit 1
+fi
+
+# This function provides help information for the script.
+function help() {
+  echo "Usage: ${0} <command> [options]"
+  echo "Commands:"
+  echo "  help            Display this help message."
+  echo "  validate_input  Validate the structure of TFVAR JSON file."
+}
+
+
+# This function is used to validate the structure of a JSON input based on a specific schema.
+# It checks for the presence and validity of the fields in the provided JSON.
+#
+# Parameters:
+#   $1 - The path to the JSON file to be validated.
+#
+# Usage: validate_input "/path/to/your/json/file.json"
+function validate_input() {
+  [[ -z "${1}" ]] && print_error "Please provide tfvar.json file as 1st argument" && return 1 || local json_tfvars="${1}" ;
+
+  # Check if the file exists
+  if [[ ! -f "${json_tfvars}" ]]; then
+    print_error "File '${json_tfvars}' does not exist."
+    exit 1
+  fi
+
+  # Validate 'aws_k8s_regions'
+  if ! jq -e '.aws_k8s_regions' "${json_tfvars}" > /dev/null; then
```

Comment:
wdyt if we can avoid repetitive ifs with something like, we also need to list each key full path, i.e. tsb_mp.cloud, etc etc basically to keep it generic

```bash

export json_tfvars="terraform.tfvars.json"

export REQUIRED_VARS=("dns_provider:aws,azure,gcp" "tsb_mp.cluster_id:" "tetrate_owner:")

for item in "${REQUIRED_VARS[@]}"; do
    option="${item%%:*}";  # Extracts everything before the colon
    allowed_values="${item##*:}"; # Extracts everything after the colon
    current_value=$(jq -r ".$option" ${json_tfvars})
    if [[ -z "$current_value" ]]; then
        echo "Missing $option in the JSON.";
    fi 
    if ! [[ -z "$allowed_values" ]] && ! [[ -z $(grep -q $allowed_values <<< $current_value) ]] ; then
        echo "$option is set to the incorrect value: $current_value, allowed values: $allowed_values \n";
    fi
done

```

Reply:
Done